executor/executor.go

```diff
@@ -472,6 +474,64 @@ func (e *CheckIndexExec) Next(ctx context.Context, chk *chunk.Chunk) error {
 	return nil
 }
 
+// ShowSlowExec represents the executor of showing the slow queries.
+// It is build from the "admin show slow" statement:
+//	admin show slow top [internal | all] N
+//	admin show slow recent N
+type ShowSlowExec struct {
+	baseExecutor
+
+	done     bool
+	ShowSlow *ast.ShowSlow
+}
+
+// Open implements the Executor Open interface.
+func (e *ShowSlowExec) Open(ctx context.Context) error {
+	if err := e.baseExecutor.Open(ctx); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+// Next implements the Executor Next interface.
+func (e *ShowSlowExec) Next(ctx context.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	if e.done {
+		return nil
+	}
+
+	dom := domain.GetDomain(e.ctx)
+	slowQueries := dom.ShowSlowQuery(e.ShowSlow)
+	for _, slow := range slowQueries {
```

Comment:
the number of rows appended to `chk` should be smaller than `max_chunk_size`

Reply:
So if the result is larger than `max_chunk_size`, it has to be split into multiple batches?

Reply:
Yes.

Reply:
we have to use a cursor to record how many slow queries have been returned to the client.